pkg/windows/cni.go

```diff
@@ -0,0 +1,25 @@
+// +build windows
+
+package windows
+
+import (
+	"context"
+	"fmt"
+
+	daemonconfig "github.com/rancher/k3s/pkg/daemons/config"
+)
+
+type Cni interface {
+	Setup(ctx context.Context, nodeConfig *daemonconfig.Node) (*CniConfig, error)
+	Start(config *CniConfig) error
+}
+
+// NewCni returns the correct implementation based on the cniType specified.
+func NewCni(cniType string) (Cni, error) {
```

Comment:
It's an antipattern in Go to return interfaces from constructors. We should be returning concrete types and define things with an interface where the concrete type would be assigned.